### PR TITLE
Chmod flag for add and copy

### LINKED ISSUE #2961: Dockerfile ADD and COPY support --chmod=755 flag
See https://github.com/moby/buildkit/pull/1492

This is useful, it would be nice if buildah supported this too.

### PR DESCRIPTION
**Pull request by jmguzik**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
Adds chmod flag to ADD and COPY
#### How to verify it
Used bats files, but open to suggestions

#### Which issue(s) this PR fixes:
Fixes #2961

#### Does this PR introduce a user-facing change?
Yes

```
Support added for --chmod flag to ADD/COPY in Dockerfile
Support added for --chmod flag to buildah cli ADD/COPY
```

### CODE REVIEW

**Comment by rhatdan** [comment]
Thanks @jmguzik 

Can you add --chmod to `buildah copy` and `buildah add`?

To complete the change.

**Comment by jmguzik** [comment]
@rhatdan yes, that was my intention, but in separate PR. No problem though, I can work on it here.

**Comment by jmguzik** [comment]
For info: issue https://github.com/openshift/imagebuilder/pull/184 is a kind of a blocker. Until it gets resolved this PR won't move further.
buildah copy and build add are implemented, dockerfile handling needs changes in imagebuilder.

**Comment by jmguzik** [comment]
@rhatdan probably release of imagebuilder is needed?

**Comment by rhatdan** [comment]
Yes I am trying to get one more fix for imagebuilder done, and then we can trigger a release.

**Comment by rhatdan** [comment]
https://github.com/openshift/imagebuilder/releases/tag/v1.2.0

**Comment by rhatdan** [comment]
LGTM
@vrothberg @saschagrunert @nalind @TomSweeneyRedHat PTAL

File: docs/buildah-add.md [hunk 1]
```diff
@@ -23,6 +23,10 @@ Defaults to false.
 Note: You can also override the default value of --add-history by setting the
 BUILDAH\_HISTORY environment variable. `export BUILDAH_HISTORY=true`
 
+**--chmod** *permissions*
+
+Sets the access permissions of the destination content.
```
Comment (vrothberg) [inline docs/buildah-add.md:28]: Can we specify the input format?  A `--chmod +x`, for instance, won't work.

**Review by vrothberg — COMMENTED** [review]
Just one doc nit. Other than that LGTM.

Please remove the "Draft" state before we can merge.

**Comment by jmguzik** [comment]
@rhatdan @vrothberg this is anything else but not LGTM :)

It is 1/2 of a feature. I will implement today/tomorrow Dockerfile handling (FYI, it works, writing tests right now).
I can divide it to 2 PRs (doing for this one only buildah cli) or 2 commits (in this PR). 

Right now, this code is handling only cli arguments. Second will be about Dockerfile.

File: docs/buildah-add.md (hunk 1 above)
Comment (jmguzik) [reply docs/buildah-add.md]: done

**Comment by jmguzik** [comment]
@rhatdan @vrothberg 
If all tests will be OK, then it's ready to be reviewed.

**Comment by rhatdan** [comment]
LGTM Except, you need to add these flags to completions. contrib/completions/bash/buildah
@containers/podman-maintainers PTAL

**Comment by rhatdan** [comment]
While you are in there, could you also fix the -chown under buildah_add, it is missing a "-"

**Comment by jmguzik** [comment]
> While you are in there, could you also fix the -chown under buildah_add, it is missing a "-"

Done

**Comment by rhatdan** [comment]
Great job @jmguzik 
/approve
/lgtm

File: docs/buildah-copy.md [hunk 2]
```diff
@@ -21,6 +21,10 @@ Defaults to false.
 Note: You can also override the default value of --add-history by setting the
 BUILDAH\_HISTORY environment variable. `export BUILDAH_HISTORY=true`
 
+**--chmod** *permissions*
+
+Sets the access permissions of the destination content. Accepts the numerical format.
```
Comment (TomSweeneyRedHat) [inline docs/buildah-copy.md:26]: It would be nice to add an example in the example section here and on the add man page.

Comment (jmguzik) [reply docs/buildah-copy.md:26]: Sure, will open a PR to complement tomorrow.

**Comment by mikepurvis** [comment]
I think this is still missing from `buildah bud`, with buildah version 1.19.8:

```
STEP 5: COPY --chmod=644 mycorp.crt   usr/local/share/ca-certificates/extra/mycorp.crt
error building at STEP "COPY --chmod=644 mycorp.crt usr/local/share/ca-certificates/extra/mycorp.crt": COPY only supports the --chown=<uid:gid> and the --from=<image|stage> flags
```

**Pull request merged**